scripts/gen_connector_charm.py

```diff
@@ -776,6 +776,104 @@ def gen_mitre_connector(location: pathlib.Path, version: str) -> None:
+                "type": "string",
+                "optional": False,
+            },
+            "nti-api-key": {
+                "description": "The NTI API key.",
+                "type": "string",
+                "optional": False,
+            },
+            "nti-tlp": {
+                "description": "TLP Marking for all data imported from NTI, possible values- white, green, amber, amber+strict, red.",
+                "type": "string",
+                "optional": False,
+            },
+            "connector-queue-threshold": {
+                "description": "(optional) Used to determine the limit (RabbitMQ) in MB at which the connector must go into buffering mode.",
+                "type": "int",
+                "default": 500,
+                "optional": True,
+            },
+            "nti-create-ioc": {
+                "description": "(optional) If true then indicators will be created for each import.",
+                "type": "boolean",
+                "optional": True,
+            },
+            "nti-create-ip": {
+                "description": "(optional) If true then IP observables will be created for each import.",
+                "type": "boolean",
+                "optional": True,
+            },
+            "nti-create-domain": {
+                "description": "(optional) If true then Domain observables will be created for each import.",
+                "type": "boolean",
+                "optional": True,
+            },
+            "nti-create-file": {
+                "description": "(optional) If true then File observables will be created for each import.",
+                "type": "boolean",
+                "optional": True,
+            },
+            "nti-create-url": {
+                "description": "(optional) If true then URL observables will be created for each import.",
+                "type": "boolean",
+                "optional": True,
+            },
+        },
+        charm_override=textwrap.dedent(
+            """\
+            VALID_LOG_LEVELS = {"debug", "info", "warn", "error"}
+    
+            def _gen_env(self) -> dict[str, str]:
+                import logging
+                logger = logging.getLogger(__name__)
+                env = super()._gen_env()
+
+                # Force nti
+                env["CONNECTOR_SCOPE"] = "nti"
+
+                # Check log level value, logging and setting to default if there is a misconfiguration
+                log_level = env.get("CONNECTOR_LOG_LEVEL", "info")
+                if log_level not in self.VALID_LOG_LEVELS:
+                    logger.warning(
+                        f"CONNECTOR_LOG_LEVEL '{log_level}' is invalid, using 'info' instead."
+                    )
+                    log_level = "info"
+                env["CONNECTOR_LOG_LEVEL"] = log_level
+
+                return env
+            """
+        ),
+        generate_entrypoint="echo ' #opencti-connector-nti' >> entrypoint.sh",
```

Comment:
```suggestion
        install_location="NTI-connector",
```

It should be `install_location` because the `entrypoint.sh` in the source repository points to `/opt/opencti/NTI-connector/src/main.py`.

Reply:
Thank you for that, didn't notice that config was there